zebra-network/src/meta_addr.rs

```diff
@@ -488,11 +488,17 @@ impl MetaAddr {
     ///
     /// Returns `None` if this `MetaAddr` should not be sent to remote peers.
     pub fn sanitize(&self) -> Option<MetaAddr> {
+        if !self.last_known_info_is_valid_for_outbound() {
+            return None;
+        }
```

Comment:
This change makes sure we don't share invalid addresses with our peers.
(But we want to keep some of them internally, so we don't reconnect to them.)